x/data/server/keys.go

```diff
@@ -21,19 +20,29 @@ func CIDBase64String(cid []byte) string {
 }
 
 func CIDSignerKey(cidStr string, signer string) []byte {
-	return append([]byte{CIDSignerPrefix}, []byte(fmt.Sprintf("%s/%s", cidStr, signer))...)
+	key := CIDSignerIndexPrefix(cidStr)
+	key = append(key, signer...)
+	return key
 }
 
 func CIDSignerIndexPrefix(cidStr string) []byte {
-	return append([]byte{CIDSignerPrefix}, []byte(fmt.Sprintf("%s/", cidStr))...)
+	key := []byte{CIDSignerPrefix}
+	key = append(key, cidStr...)
+	key = append(key, 0)
+	return key
 }
 
-func SignerCIDKey(signer string, cidStr string) []byte {
-	return append([]byte{SignerCIDPrefix}, []byte(fmt.Sprintf("%s/%s", signer, cidStr))...)
+func SignerCIDKey(signer string, cid []byte) []byte {
+	key := SignerCIDIndexPrefix(signer)
+	key = append(key, cid...)
+	return key
 }
 
 func SignerCIDIndexPrefix(signer string) []byte {
-	return append([]byte{SignerCIDPrefix}, []byte(fmt.Sprintf("%s/", signer))...)
+	key := []byte{SignerCIDPrefix}
+	key = append(key, signer...)
+	key = append(key, 0)
```

Comment:
Any particular reason for switching to `0` instead of `/` as a delimeter? If we expect users to ever be directly querying KVStore via tendermint RPC, then it does probably make sense for us to think of this as an API surface.

If we do switch here, should we also change other prefix deliminters (e.g. ecocredit's batchDenom prefixes also use `/` [here](https://github.com/regen-network/regen-ledger/blob/master/x/ecocredit/server/msg_server.go#L65))?

Reply:
Since both are strings, `0` is guaranteed to not be a string as it's the null terminator. `'/'` is a valid ASCII character

Reply:
So this is a more generic way to separate two strings in an index key

Reply:
I saw somewhere a note to use `0` (probably in an ADR). It will be worth to add a documentation to this function and pointing to a specification if available.